### PR TITLE
A single content type to handle all json data

### PR DESCRIPTION
**Pull request by fmizzell**

### CODE REVIEW

**Comment by thierrydallacroce** [comment]
Site spun up with single Data content type, with the usual JSON Metadata field, as well as a new Type field:

Screen Shot 2019-04-10 at 10 47 57 AM
Screen Shot 2019-04-10 at 10 48 03 AM

**Comment by thierrydallacroce** [comment]
Regarding code review, should we be concerned with the multiple mentions of "Dataset" or "dataset" in what appears to be the old context, i.e. more than a value in the new Type field. For example, in dkan_api, within the controller class name, the url of the Rest API, or the permission name?

File: modules/custom/interra_api/src/Search.php [hunk 1]
```diff
@@ -13,18 +13,19 @@ public function formatDocs($docs) {
...
   }
-  public function formatSearchDoc($node) {
+  public function formatSearchDoc($value) {
     $formatted = new \stdClass();
-    $value = $node->get('field_json_metadata')->getValue()[0]['value'];
-    $doc = $this->dereference(json_decode($value));
-    $formatted->doc = $doc;
-    $formatted->ref = \Drupal::service('path.alias_manager')->getAliasByPath('/node/'. $node->id());
+    $formatted->doc = $value;
+    $formatted->ref = \Drupal::service('path.alias_manager')->getAliasByPath('/node/x');
```
[the remark this answers is not in the document]
Comment (fmizzell) [reply modules/custom/interra_api/src/Search.php]: :) we lose that context of "node" by interacting with the API directly. I am pretty certain that the frontend does not use that "ref" property anywhere. Maybe just deleting it is better than leaving some random code in there?

**Comment by fmizzell** [comment]
@thierrydallacroce The concept of dataset from the user's perspective and the API perspective still exists. There might be some contexts in which dataset is not an appropriate name anymore, but we will have to look at those one by one.

File: modules/custom/interra_api/src/Search.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (thierrydallacroce) [reply modules/custom/interra_api/src/Search.php]: Definitely a case of my focusing too much on the `/x` and not seeing the forest for the trees. So, whatever you feel makes more sense is fine by me.

**Pull request merged**